### PR TITLE
Added Nag Dialog for Prisoners of War Outside of Contracts

### PR DESCRIPTION
**Pull request by IllianiCBT**
### Current Implementation
None

### Problem
I am definitely not the only person who has done this: end the contract, enjoy the spoils of war, get halfway to Galatea only to remember you have a couple hundred prisoners of war still hanging about. You're not about to return to the contract planet so you just dump them out into space. Everyone is sad.

### Solution
On a new day, if not on a contract, the system will check whether any of the users' active personnel are prisoners. If any prisoners are found the nag dialog is triggered.

<p align=center>
image
image
</p>

### Update
I removed the check for Bondsmen, as these can still be assigned to roles they don't really have the same issue as Prisoners.

### CODE REVIEW

**Comment by AaronGullickson** [comment]
One thing to keep in mind is that people use MekHQ for all kinds of things not just AtB, so we should be careful about assuming purpose and intent of having prisoners. I don't love having this nag if I *want* to keep prisoners for some reason, but I suppose it can be dismissed easily enough.

**Comment by IllianiCBT** [comment]
> One thing to keep in mind is that people use MekHQ for all kinds of things not just AtB, so we should be careful about assuming purpose and intent of having prisoners. I don't love having this nag if I _want_ to keep prisoners for some reason, but I suppose it can be dismissed easily enough.

Valid point. My thinking is that, in instances where users _want_ to keep their prisoners, the nag is easy to suppress and re-enable at a later date. Much like unmaintained units.

**Comment by IllianiCBT** [comment]
> Yeah, that's fine. If you can just clear the other excess parenthesis, I can approve and merge.

Done (assuming there wasn't others). I also changed `hasPrisoners()` to private, as there was no reason for it to be public.

**Review by AaronGullickson — APPROVED** [review]

File: MekHQ/src/mekhq/gui/dialog/nagDialogs/PrisonersNagDialog.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import mekhq.campaign.personnel.Person;
+import mekhq.gui.baseComponents.AbstractMHQNagDialog;
+
+import javax.swing.*;
+
+public class PrisonersNagDialog extends AbstractMHQNagDialog {
+    public static boolean hasPrisoners (Campaign campaign) {
+        if (!campaign.hasActiveContract()) {
+            for (Person p : campaign.getActivePersonnel()) {
+                if ((p.getPrisonerStatus().isCurrentPrisoner())) {
```
[the remark this answers is not in the document]
Comment (IllianiCBT) [reply MekHQ/src/mekhq/gui/dialog/nagDialogs/PrisonersNagDialog.java]: Done :)

**Pull request merged**